plugins/kotlin/fir/src/org/jetbrains/kotlin/idea/completion/weighers/PreferGetSetMethodsToPropertyWeigher.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+package org.jetbrains.kotlin.idea.completion.weighers
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementWeigher
+import com.intellij.codeInsight.lookup.WeighingContext
+import com.intellij.openapi.util.Key
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtPropertySymbol
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtSymbol
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.UserDataProperty
+
+internal object PreferGetSetMethodsToPropertyWeigher {
+    const val WEIGHER_ID = "kotlin.preferGetSetMethodsToProperty"
+    private var LookupElement.propertyName by UserDataProperty(Key<Name>("KOTLIN_PROPERTY_NAME"))
+
+    fun KtAnalysisSession.addWeight(lookupElement: LookupElement, symbol: KtSymbol) {
+        lookupElement.propertyName = (symbol as? KtPropertySymbol)?.name
+    }
+
+    object Weigher : LookupElementWeigher(WEIGHER_ID, false, true) {
+        override fun weigh(element: LookupElement, context: WeighingContext): Int {
+            val propertyName = element.propertyName?.asString() ?: return 0
+            val prefixMatcher = context.itemMatcher(element)
+            if (prefixMatcher.prefixMatches(propertyName)) return 0
+            val matchedLookupStrings = element.allLookupStrings.filter { prefixMatcher.prefixMatches(it) }
+            return if (matchedLookupStrings.all { it.startsWith("get") || it.startsWith("set") }) 1 else 0
```

Comment:
Actually I think we don't need to do those because this check first compares the property name with the prefix. If that matches, we return 0 directly. The comparison with `get` and `set` is only done if the property name does not match. We don't need to handle `is` because, unlike `get`, `is` is part of the property name.